bin/ping.py

```diff
@@ -230,16 +221,14 @@ def do_one(dest_addr, timeout):
     icmp = socket.getprotobyname("icmp")
     try:
         my_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, icmp)
-    except socket.error as xxx_todo_changeme:
-        (errno, msg) = xxx_todo_changeme.args
-        if errno == 1:
+    except socket.error as e:
+        if e.errno == 1:
             # Operation not permitted
-            msg = msg + (
+            msg = e.args[0] + (
                 " - Note that ICMP messages can only be sent from processes"
-                " running as root."
-            )
+                " running as root.")
             raise socket.error(msg)
-        raise # raise the original error
+        raise  # raise the original error
```

Comment:
Do you suggest removing the whole try / except?  If not, what should be in the except block?

Reply:
As far as I can tell, the `except` block's only purpose is to display the message about running as root if a permission error is raised. Since that advice isn't useful in the context of Stash, we don't need the `except` block. (Unless I'm missing something, in which case please only remove the error message modification code.)

The code inside the `try` block needs to stay of course, but not in a `try` block.